src/patch_parser.py

```diff
@@ -156,6 +156,24 @@ def diff_best_match(self, string: List[str]) -> List[str]:
         return node.diff_best_match(string)
 
 
+@dataclasses.dataclass
+class HunkParserState:
+    _number_of_edits: int
+    _deleted_lines: int
+
+    def get_number_of_deleted_lines(self):
+        return self._deleted_lines
+
+    def add_deleted_line(self):
+        self._deleted_lines += 1
+
+    def reset_deleted_lines(self):
+        self._deleted_lines = 0
+
+    def __init__(self):
```

Comment:
we don't need this, I think.

But we should be able to also do
```python
deleted_lines: int = 0 
```
up top to be more explicit instead of needing an `__init__()`